src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -63,9 +63,16 @@ public class HubEndPoint<THub, TClient> : EndPoint where THub : Hub<TClient>
 
                 using (var scope = _serviceScopeFactory.CreateScope())
                 {
-                    var hub = scope.ServiceProvider.GetService<THub>() ?? Activator.CreateInstance<THub>();
-                    InitializeHub(connection, hub);
+                    bool created;
```

Comment:
It's really a design flaw with the pattern. We need a HubActivator with a Create and Dispose like MVC's ControllerActivator (or is it the factory that disposes..)